### PR TITLE
Fixed bug in map Retry Install

### PR DESCRIPTION
**Pull request by dsimmons87**
If the download failed, the map status would be MapStatus.DownloadError,
which would cause the Install method to return immediately. I've updated
the Install method now to account for MapStatus.DownloadError.

In reference to issue #6196

### CODE REVIEW

File: OpenRA.Game/Map/MapPreview.cs [hunk 1]
```diff
@@ -418,7 +418,7 @@ public void UpdateRemoteSearch(MapStatus status, MiniYaml yaml, Action<MapPrevie
 
 		public void Install(string mapRepositoryUrl, Action onSuccess)
 		{
-			if (Status != MapStatus.DownloadAvailable || !Game.Settings.Game.AllowDownloading)
+			if (Status != MapStatus.DownloadError && (Status != MapStatus.DownloadAvailable || !Game.Settings.Game.AllowDownloading))
```
Comment (pchote) [inline OpenRA.Game/Map/MapPreview.cs]: This should be `((Status != MapStatus.DownloadError && Status != MapStatus.DownloadAvailable) || !Game.Settings.Game.AllowDownloading)`. We want to unconditionally exit here if the player has disabled map downloading.

Comment (dsimmons87) [reply OpenRA.Game/Map/MapPreview.cs]: Makes sense

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
Simple repro case for testing:
1. Download and install a custom map from the resource center.
2. Start a local dedicated server with `Server.Map=<custom map UID>`.
3. Delete the custom map from the local maps dir.  Don't quit the server after this.
4. Start the game with `Launch.Connect=localhost:1234` to launch and immediately join the server.
5. Click install.  If the install starts first time then quit the game before it completes, and retry until it fails.
6. Before the fix: pressing the retry button doesn't work and the map cannot be installed without a game restart.  After the fix: pressing the retry button works and the map will install as soon as the resource center decides to cooperate.

**Comment by reaperrr** [comment]
Looks good, but the merge commit needs to be removed before we can merge this.

**Comment by pchote** [comment]
That will automatically vanish when we do the rebase merge.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fa795c3b207bf52c09f4713c90b6cab3528433b0...d181e6b6359d523a8a8895333eca4175efb41481)

**Pull request merged**